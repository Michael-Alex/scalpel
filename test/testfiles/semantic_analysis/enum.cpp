class s
{
	enum color
	{
		RED,
		GREEN,
		BLUE
	};
};

enum direction
{
	LEFT,
	RIGHT,
};

s::color c = s::color::RED;
direction d = RIGHT;

