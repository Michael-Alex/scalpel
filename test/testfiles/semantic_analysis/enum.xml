<namespace id="n0">
 <class name="s" id="c0">
  <enum name="color" id="e0">
   <constant name="RED" value="0">
   <constant name="GREEN" value="1">
   <constant name="BLUE" value="2">
  </enum>
  <destructor access="public" defined="false">
  </destructor>
 </class>
 <enum name="direction" id="e1">
  <constant name="LEFT" value="0">
  <constant name="RIGHT" value="1">
 </enum>
 <variable name="c">
  <type>
   <enum id="e0"/>
  </type>
 </variable>
 <variable name="d">
  <type>
   <enum id="e1"/>
  </type>
 </variable>
</namespace>
