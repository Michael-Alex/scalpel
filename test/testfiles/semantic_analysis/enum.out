<namespace id="0">
 <class id="0" name="s">
  <enum id="0" name="color" access="private">
   <constant name="RED" id="0" value="0">
   <constant name="GREEN" id="1" value="1">
   <constant name="BLUE" id="2" value="2">
  </enum>
  <destructor access="public" defined="false">
  </destructor>
 </class>
 <enum id="0" name="direction">
  <constant name="LEFT" id="3" value="0">
  <constant name="RIGHT" id="4" value="1">
 </enum>
 <variable id="0" name="c">
  <type>
   <member_enum id="0"/>
  </type>
 </variable>
 <variable id="1" name="d">
  <type>
   <enum id="0"/>
  </type>
 </variable>
</namespace>
